### PR TITLE
Visualize and clarify branching model

### PR DESCRIPTION
**Pull request by mavam**
This PR adds a visualization of the branching model and clarifies how we use it.

### CODE REVIEW

**Comment by mavam** [comment]
Here's the proposed new branching model:

git branches

File: contributing.md [hunk 1]
```diff
@@ -16,37 +16,51 @@ Documentation for Tenzir projects can be found at
...
+![Branching Model](https://user-images.githubusercontent.com/53797/156560785-c7279447-63eb-4428-9a11-9c90cc03acc8.png)
+
+Our git workflow looks as follows:
 
 - The `master` branch reflects the latest state of development, and should
   always compile.
 
+- In case we need to release a hotfix, we use dedicated patch release branches.
+
+- The stable `branch` always points to the latest release that is not a release
```
Comment (dominiklohmann) [inline contributing.md]:
```suggestion
- The `stable` branch always points to the latest release that is not a release
```

File: contributing.md [hunk 2]
```diff
@@ -16,37 +16,51 @@ Documentation for Tenzir projects can be found at
...
+  candidate. CI resets it automatically. It exists so support a streamlined
```
Comment (dominiklohmann) [inline contributing.md]:
```suggestion
  candidate. It exists so support a streamlined
```

We do this manually. We definitely don't want to allow CI to force-push, like ever.

**Review by dominiklohmann — APPROVED** [review]
Some minor remarks, otherwise LGTM.

**Pull request merged**